dotcom-rendering/src/web/components/ShowMore.importable.tsx

```diff
@@ -0,0 +1,200 @@
+import { css } from '@emotion/react';
+import {
+	from,
+	neutral,
+	space,
+	visuallyHidden,
+} from '@guardian/source-foundations';
+import {
+	Button,
+	InlineError,
+	SvgCross,
+	SvgPlus,
+} from '@guardian/source-react-components';
+import { useState } from 'react';
+import type { DCRContainerPalette, FEFrontCard } from 'src/types/front';
+import { enhanceCards } from '../../model/enhanceCards';
+import { shouldPadWrappableRows } from '../lib/dynamicSlices';
+import { useApi } from '../lib/useApi';
+import { useOnce } from '../lib/useOnce';
+import { LI } from './Card/components/LI';
+import { UL } from './Card/components/UL';
+import { FrontCard } from './FrontCard';
+
+function decideButtonText({
+	isOpen,
+	loading,
+	containerTitle,
+}: {
+	isOpen: boolean;
+	loading: boolean;
+	containerTitle: string;
+}) {
+	if (isOpen && loading) return 'Loading';
+	if (isOpen) return `Less ${containerTitle}`;
+	return `More ${containerTitle}`;
+}
+
+type Props = {
+	containerTitle: string;
+	path: string;
+	containerId: string;
+	showAge: boolean;
+	containerPalette?: DCRContainerPalette;
+};
+
+export const ShowMore = ({
+	containerTitle,
+	path,
+	containerId,
+	showAge,
+	containerPalette,
+}: Props) => {
+	function findCardLinks() {
+		const containerNode = document.getElementById(containerId);
+		const containerLinksArray = Array.from(
+			containerNode?.querySelectorAll('a') ?? [],
+		);
+		return containerLinksArray.map(
+			(Element) => Element.attributes.getNamedItem('href')?.value,
+		);
+	}
+
+	const [cardLinks, setCardLinks] = useState<string[]>([]);
+
+	useOnce(() => {
+		const linksArray: string[] = findCardLinks().filter(
+			(item): item is string => !!item,
```

Comment:
are we happy with inlined type guards like this, from a readability perspective? the purpose here is just to tell tsc something that it should already know but which it doesn't because of some quirks around filtering.